src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -116,37 +123,52 @@ public void onFailure(Throwable caught) {
 
 			@Override
 			public void onSuccess(Void result) {
-				initializeFromExistingSessionCookie(callback);
+				initializeFromExistingAccessTokenCookie(callback);
 			}
 		});
 	}
 
 	/**
-	 * Session cookie should be set before this call
+	 * Access token cookie should be set before this call
 	 * 
 	 * @param callback
 	 */
-	public void initializeFromExistingSessionCookie(AsyncCallback<UserProfile> callback) {
-		userAccountService.getCurrentUserSessionData(new AsyncCallback<UserSessionData>() {

```

Comment:
UserSessionData is going away (it's session specific), so I needed to figure out how to do this in the new world during initialization.  First step is to get the access token (if there is one).  Call will fail if there is none, or if the access token is invalid (which is a new change to the InitSessionServlet doGet(), you can find below). 
If the access token is present (and valid), then get the current user profile.  If this fails because the terms of use have not been signed, then send the user to the ToU (like we did previously).